acceptance/tests/agent_sends_json_report_for_cached_catalog.rb

```diff
@@ -0,0 +1,34 @@
+test_name "C100533: Agent sends json report for cached catalog" do
+  skip_test 'requires a master' if master.nil?
+
+  tag 'risk:medium',
+      'audit:medium',
+      'audit:integration',
+      'server'
+
+  with_puppet_running_on(master, :main => {}) do
+    expected_format = 'json'
+
+    step "Perform agent run to ensure that catalog is cached" do
+      agents.each do |agent|
+        on(agent, puppet('agent', '-t', "--server #{master}"), :acceptable_exit_codes => [0,2])
+      end
+    end
+
+    step "Ensure agent sends #{expected_format} report for cached catalog" do
+      agents.each do |agent|
+        on(agent, puppet('agent', '-t',
+                         "--server #{master}",
+                         '--http_debug'), :acceptable_exit_codes => [0,2]) do |res|
+          found_format = false
+          # Expected content-type should be in the headers of the
+          # HTTP report payload being PUT to the server by the agent.
+          found_format = true if res.stderr =~ /<- "PUT \/puppet\/v[3-9]\/report.*Content-Type: .*\/#{expected_format}/
+          fail_test("Report was not submitted in #{expected_format} format") unless found_format == true
```

Comment:
Why bother with the found_format variable and logic with this simple case.

unless res.stderr =~ /<- "PUT \/puppet\/v[3-9]\/report.*Content-Type: .*\/#{expected_format}/
    fail_test(...)
end

I can live either way.